nkms_eth/blockchain.py

```diff
@@ -4,46 +4,70 @@
+    ropsten: Connects to the public ethereum ropsten testnet via geth.
+    tester: Uses an ephemeral in-memory chain backed by pyethereum.
+    testrpc: Uses an ephemeral in-memory chain backed by pyethereum.
+    temp: Local private chain whos data directory is removed when the chain is shutdown. Runs via geth.
 
-def project():
-    # Hardcoded the config for the project
-    # It will read user-specific configs also which may override it
-    if not hasattr(_project, 'project'):
+    """
+
+    network = ''  # 'mainnetrpc'
+    python_project_name = 'nucypher-kms'
+    _project = threading.local()
+
+    # This config is persistent and is created in user's .local directory
+    registrar_path = join(appdirs.user_data_dir(python_project_name), 'registrar.json')
+
+    def __init__(self, project_name='nucypher-kms', timeout=60):
+
+        # Populus project config
         project_dir = join(dirname(abspath(nkms_eth.__file__)), 'project')
         project = populus.Project(project_dir)
-        project.config['chains.mainnetrpc']['contracts']['backends']['JSONFile']['settings']['file_path'] = REGISTRAR_PATH
-        _project.project = project
-    return _project.project
+        project.config['chains.mainnetrpc.contracts.backends.JSONFile.settings.file_path'] = self.registrar_path
+
+        self.project_name = project_name
+        self.timeout = timeout
+        self.project_dir = project_dir
+        self._project.project = project
+        self._project.chain = self._project.project.get_chain(self.network).__enter__()
+
+    def __repr__(self):
+        class_name = self.__class__.__name__
+        r = "{}(network={}, project_name={}, timeout={})"
+        return r.format(class_name, self.network, self.project_name, self.timeout)
 
+    def __str__(self):
+        class_name = self.__class__.__name__
+        return "{} {}:{}".format(class_name, self.network, self.project_name)
 
-def get_chain(name=None):
-    return project().get_chain(name or DEFAULT_NETWORK)
+    def disconnect(self):
+        self._project.chain.__exit__(None, None, None)
 
+    @property
+    def chain(self):
+        return self._project.chain
 
-def chain(name=None):
-    if not hasattr(_project, 'chain'):
-        _project.chain = get_chain(name).__enter__()
-    return _project.chain
+    @property
+    def web3(self):
+        return self._project.chain.web3
 
+    def get_contract(self, name):
+        """ Gets an existing contract or returns an error """
+        return self._project.chain.provider.get_contract(name)
 
-def disconnect():
-    _project.chain.__exit__(None, None, None)
-    if hasattr(_project, 'project'):
-        delattr(_project, 'project')
-    if hasattr(_project, 'chain'):
-        delattr(_project, 'chain')
-    if hasattr(_project, 'web3'):
-        delattr(_project, 'web3')
+    def wait_time(self, wait_hours, step=50):
+        end_timestamp = self.web3.eth.getBlock(self.web3.eth.blockNumber).timestamp + wait_hours * 60 * 60
```

Comment:
Maybe a comment here to clarify when this is?  Or maybe it can be split into two lines to achieve that same goal?